### DISCUSSION TITLE
Any way to mutate a value before saving?

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#546] [Help] [answered]
I am using the `User` model in my app and want to provide my adminstrators the ability to be able to reset `User`s passwords if needed. I can see that an obfuscated password field is available but this just stores the value as plain text (I didn't really expect it to `Hash` the password for me but thought I'd check!). Is there any way to hook into the mechanism when a field is saved back to the DB?

**Comment by danharrin** [comment, marked as answer]
This is easy in v2, but you should use the hooks in v1 to do this. You need to use `beforeCreate()` and `beforeSave()`, to set `$this->record['password']` with the hashed password.

https://filamentadmin.com/docs/resources

**Reply by titantwentyone** [reply]
Thanks Dan and apologies - I must have skipped that part. I seem to have it working with a small caveat. My form fields are:

```
return $form
    ->schema([
        Components\TextInput::make('name')->autofocus()->required(),
        Components\TextInput::make('password')
            ->password()
            ->confirmed(),
        Components\TextInput::make('passwordConfirmation')
            ->password()
    ]);
```
and in my `EditRecord` class:
```
protected function beforeSave()
{
    $this->record['password'] = Hash::make($this->record['password']);

    unset($this->record['passwordConfirmation']);
}
```
This works perfectly but I had to unset the `passwordConfirmation`. Not doing so results in an attempt to save `passwordConfirmation` to the record.

**Reply by fauzie811** [reply]
How do I implement this in a simple (modal) resource?

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/admin/resources/listing-records#customizing-data-before-saving

**Comment by danharrin** [comment]
We do the same [here](https://github.com/laravel-filament/filament/blob/main/src/Resources/UserResource/Pages/EditUser.php#L24).

In v2, you will just be able to use `dehydrated(false)` on confirmation field to do the same thing automatically.

**Reply by titantwentyone** [reply]
Ah. Good to know I was doing the right thing. Thanks!
